lib/src/requester/requester.dart

```diff
@@ -130,7 +131,19 @@ class Requester extends ConnectionHandler {
         listener.cancel();
         listener = null;
       }
+      if (to != null && to.isActive) {
+        to.cancel();
+        to = null;
```

Comment:
Not sure I like the assignment to null. Instead I'd rather reuse the same timer reference. It'd clean the code up for null checks without changing the behaviour too much. Any thoughts @butlermatt ?

Reply:
It's a local variable to the method so the variable will never be reused, With every call a new variable is allocated for it. 
Either way the timer still will require null checks as it may not always be set (eg if no timeout duration is ever passed in, then the timer (to) will be null). Mainly it was just just force a garbage collection of the the variable if for some reason the closure wasn't gc'd somewhere else (the one passed to subscribe).